qualitified-crm-core/src/test/java/org/qualitified/crm/listener/TestPreCreate.java

```diff
@@ -0,0 +1,93 @@
+package org.qualitified.crm.listener;
+
+import com.google.inject.Inject;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.nuxeo.ecm.automation.OperationException;
+import org.nuxeo.ecm.automation.test.AutomationFeature;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.event.EventService;
+import org.nuxeo.ecm.core.event.impl.EventListenerDescriptor;
+import org.nuxeo.ecm.platform.test.PlatformFeature;
+import org.nuxeo.runtime.test.runner.Deploy;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.Assert.*;
+
+@RunWith(FeaturesRunner.class)
+@Features({ PlatformFeature.class, AutomationFeature.class})
+@Deploy({"studio.extensions.crm"})
+public class TestPreCreate {
+
+    protected final List<String> events = Arrays.asList("emptyDocumentCreated");
+
+    @Inject
+    protected EventService s;
+
+    @javax.inject.Inject
+    CoreSession coreSession;
+
+    @Test
+    public void listenerRegistration() {
+        EventListenerDescriptor listener = s.getEventListener("precreate");
+        assertNotNull(listener);
+        assertTrue(events.stream().allMatch(listener::acceptEvent));
+    }
+
+    @Test
+    public void runPreCreate() throws OperationException{
+        DocumentModel script = coreSession.createDocumentModel("/Admin/Scripts", "FilePreCreate","ScriptNote");
+        script=coreSession.createDocument(script);
```

Comment:
@MabroukaJRAD You should add a real test here.
For Ex:
script.setPropertyValue("note:mime_type", "text/plain");
script.setPropertyValue("note:note", "input['dc:description']='value set via script';");
...
and verify that the "dc:description" metadata of the file document contains 'value set via script' after the document is created.